nipype/algorithms/confounds.py

```diff
@@ -1158,16 +1231,24 @@ def compute_noise_components(imgseries, mask_images, num_components,
     period_cut: minimum period (in sec) for DCT high-pass filter
     repetition_time: time (in sec) between volume acquisitions
 
-    returns:
-
-    components: a numpy array
-    basis: a numpy array containing the (non-constant) filter regressors
-
+    Returns
+    -------
+    components: numpy array
+        Numpy array containing the requested set of noise components
+    basis: numpy array
+        Numpy array containing the (non-constant) filter regressors
+    metadata: OrderedDict{str: numpy array}
+        Dictionary of eigenvalues, fractional explained variances, and
+        cumulative explained variances.
     """
     components = None
     basis = np.array([])
-    for img in mask_images:
-        mask = img.get_data().astype(np.bool)
+    if components_criterion == 'all':
+        components_criterion = -1
+    if not mask_names:
+        mask_names = range(len(mask_images))
+    for i, img in zip(mask_names, mask_images):
```

Comment:
```suggestion
    for name, img in zip(mask_names, mask_images):
```
Then fix subsequent appearances of ``i`` later on. You overload it in the end, I wonder if that is what is confusing Python 2.7.